src/libostree/ostree-sysroot-deploy.c

```diff
@@ -1332,11 +1332,15 @@ fsfreeze_thaw_cycle (OstreeSysroot *self,
       /* Do a freeze/thaw cycle; TODO add a FIFREEZETHAW ioctl */
       if (ioctl (rootfs_dfd, FIFREEZE, 0) != 0)
         {
-          /* Not supported, or we're running in the unit tests (as non-root)?
+          /* Not supported, we're running in the unit tests (as non-root), or
+           * the filesystem is already frozen (EBUSY).
            * OK, let's just do a syncfs.
            */
-          if (G_IN_SET (errno, EOPNOTSUPP, EPERM))
+          if (G_IN_SET (errno, EOPNOTSUPP, EPERM, EBUSY))
             {
+              /* Warn if the filesystem was already frozen */
+              if (errno == EBUSY)
+                g_warning ("Filesystem already frozen, falling back to syncfs");
```

Comment:
I'm not a big fan of random spew from libraries.  (What spew we do I'd prefer to use `sd_journal_print` and not stderr).   Let's drop the `g_warning()` here?

I'd been thinking to at some point gather stats on how long `syncfs()` takes etc (and eventually a signal so clients can know we're doing it), so we could also log whether we used `FIFREEZE` or `syncfs()` then too.

Reply:
I suggest s/`g_warning`/`g_debug`/ instead of just dropping it.

Reply:
Sure. So, no message or add a journal ID for one? Also, same for the warning when thawing?

Reply:
`g_debug` is OK by me for now.

Reply:
👍 reworking